ament_copyright/ament_copyright/parser.py

```diff
@@ -44,6 +45,7 @@ def __init__(self, filetype, path):
         self.path = path
         self.exists = os.path.exists(path)
         self.content = None
+        self.license_identifier = UNKNOWN_IDENTIFIER
```

Comment:
Right, I guess that is the point I disagree with :).  While that base class itself doesn't have a license, all of its sub-classes do (they all call identify_license()).  Thus it seems to be something common to all of them, and seems to make sense in the base class.  It's similar in many respects to the identify_license() method, in that it is a convenient place to not duplicate the code, rather than it being somehow intrinsic to the base class.